lib/web_ui/lib/src/engine/html/surface.dart

```diff
@@ -535,7 +549,13 @@ abstract class PersistedSurface implements ui.EngineLayer {
   /// The default implementation is identity transform. Concrete
   /// implementations may override this getter to supply a different transform.
   Matrix4? get localTransformInverse =>
-      _localTransformInverse ??= Matrix4.identity();
+      _localTransformInverse ??= defaultLocalTransformInverse;
+  set localTransformInverse(Matrix4? value) {
+    _localTransformInverse = value;
+  }
+
+  /// The default value to be used whenever [_localTransformInverse] is null.
+  Matrix4? get defaultLocalTransformInverse => Matrix4.identity();
```

Comment:
This is probably the riskiest change in this PR. Please take a careful look @ferhatb

Reply:
Why not keep this as : 
Matrix4? localTransformInverse;
Matrix4? defaultLocalTransformInverse => localTransformInverse;

Since the only use site of defaultLocalTransformInverse getter does check for null and identity, we would not have to allocate an identity matrix on most surfaces.